airflow/cli/commands/dag_command.py

```diff
@@ -343,6 +344,24 @@ def dag_list_dags(args):
     )
 
 
+@cli_utils.action_cli
+@suppress_logs_and_warning
+@provide_session
+def dag_details(args, session=NEW_SESSION):
+    """Get DAG details given a DAG id"""
+    dag = DagModel.get_dagmodel(args.dag_id, session=session)
+    if not dag:
+        raise SystemExit(f"DAG: {args.dag_id} does not exist in 'dag' table")
+    dag_detail = dag_schema.dump(dag)
+    for key, value in dag_detail.items():
+        if isinstance(value, dict):
+            print(f"\t{key}:")
+            for subkey, subvalue in value.items():
+                print(f"\t\t{subkey}: {subvalue}")
+        else:
+            print(f"\t{key}: {value}")
```

Comment:
You are trying to implement a new logic for yaml output, and this doesn't support other types (json, table, plain).
Instead you can use `AirflowConsole.print_as` method, something like:
```suggestion
   AirflowConsole().print_as(
        data=[dag_detail],
        output=args.output,
    ) 
```

Reply:
Yep. All the CLI commands should use the same formatting values.

Reply:
Are you sure you want to do that @potiuk, @hussein-awala ? The default fallback when no --output option is provided is table -> so args.output becomes table by default, the SimpleTable isn't able to render such a large table properly due to the number of parameters and the length of these fields - most devices won't be able to display this output in a comprehensible way.
To standardise the approach, I've switched to AirflowConsole(), with a caveat that "table" isn't allowed as an output option - if it's received "yaml" output is displayed instead.

<h2> Table/Default Format </h2>
![table](https://i.ibb.co/nLSJdth/Screenshot-2023-04-02-at-9-58-34-PM.png)
<br>

<h2> JSON Format </h2>
![json](https://i.ibb.co/hCF6Vbj/Screenshot-2023-04-02-at-9-55-14-PM.png)
<br>

<h2> YAML Format </h2>
![yaml](https://i.ibb.co/7KWVPDg/Screenshot-2023-04-02-at-9-54-43-PM.png)
<br>

<h2> Plain Format </h2>
![plain](https://i.ibb.co/4P4rmNQ/Screenshot-2023-04-02-at-9-56-08-PM.png)
<br>

Reply:
Can you make a table support with replaced columns and rows ? I think that one will be much better.

Reply:
Ooo so essentially a transpose of the current table?
Let me know if I understand it currently, if initially the columns were scheduler_lock, is_active with Row1 being [], False -> now the rows will be scheduler_lock, is_active with Column1 having [], False?
Also, do we want this to be a new output type like --ttable? (Extra t for transposed)

Reply:
Yes. Transposed. No we do not need to have special type. We are free to output it in the form that will be readable. Table format is for human to look at, so having it laid out in the way that will be easier for human just makes sense as the only way of outputting it in --table form.

Reply:
Got it, we will still be calling AirflowConsole, so I'd need to generalize the logic, can I say if the number of columns is greater than 8 we should switch to transposed form?

Reply:
Better change the data passed to it and transpose it first without changing it in general. I think that is simpler.

Reply:
Convert attribute of the dag_detail objects into rows simply.

Reply:
I like the idea of the transposed table, it's simple with no need to update the `print_as` method:
```python
if args.output == "table":
    data = [
        {
            "property_name": key,
            "property_value": value
        }
        for key, value in dag_detail.items()
    ]
else:
    data = [dag_detail]
```

Reply:
precisely.

Reply:
Changes done, and everything looks fine now for table (all other outputs in my previous comment) - 
![image](https://user-images.githubusercontent.com/56473490/229546961-726b0b95-2a33-4bee-9855-4186737aabc6.png)

Reply:
As you can see in my previous comments, table & plain both formats were not rendered properly for the large number of attributes, so I've added checks for both of them.

Reply:
Nice. That's what I thought will be nicer :)